core/trino-main/src/main/java/io/trino/execution/QueryExecution.java

```diff
@@ -86,23 +87,21 @@
     }
 
     /**
-     * Output schema and buffer URIs for query.  The info will always contain column names and types.  Buffer locations will always
-     * contain the full location set, but may be empty.  Users of this data should keep a private copy of the seen buffers to
-     * handle out of order events from the listener.  Once noMoreBufferLocations is set the locations will never change, and
-     * it is guaranteed that all previously sent locations are contained in the buffer locations.
+     * The info will always contain column names and types.
+     * Once noMoreInputs is set the {@link #drainInputs(Consumer)} method is guaranteed not to produce any new inputs.
```

Comment:
I think this is not accurate or I am not reading it correctly.
It is possible to have `QueryOutputInfo` with `noMoreInputs==true` and still non-empty `inputsQueue`, right?
As I read the comment it feels that it says it should not happen.

Reply:
Good catch. Basically what I was trying to say that once `noMoreInputs` the queue inside should't contain any new splits in any subsequent calls. Let me try to rephrase.